CRM/Contact/BAO/Contact.php

```diff
@@ -175,7 +175,7 @@ public static function add(&$params) {
     // Since hash was required, make sure we have a 0 value for it (CRM-1063).
     // @todo - does this mean we can remove this block?
     // Fixed in 1.5 by making hash optional, only do this in create mode, not update.
-    if ((!array_key_exists('hash', $contact) || !$contact->hash) && !$contact->id) {
+    if ((!property_exists($contact, 'hash') || !$contact->hash) && !$contact->id) {
```

Comment:
The other option would be isset() but i feel like that might be taken care of by the other parts of the if